yargs.js

```diff
@@ -287,6 +291,12 @@ function Yargs (processArgs, cwd, parentRequire) {
     return options.demanded
   }
 
+  var recommendCommands
+  self.recommendCommands = function (value) {
+    recommendCommands = !!value
```

Comment:
I'd be tempted to interpret `. recommendCommands()` (with no arg) as enabling this functionality. The only other API method we have that only takes a boolean/truthy/falsey value is `detectLocale` which is enabled by default (so calling `.detectLocale()` there doesn't really make sense).

Perhaps we could change this line to be `recommendCommands  = value !== false` instead, so that it's enabled for no args or any arg that is not explicitly `false`.